pkg/proxy/util/nodeport_addresses.go

```diff
@@ -84,27 +86,21 @@ func (npa *NodePortAddresses) String() string {
 	return fmt.Sprintf("%v", npa.cidrStrings)
 }
 
-// GetNodeAddresses returns all matched node IP addresses for npa's IP family. If npa's
-// CIDRs include "0.0.0.0/0" or "::/0", then that value will be returned verbatim in
-// the response and no actual IPs of that family will be returned. If no matching IPs are
-// found, GetNodeAddresses will return an error.
+// MatchAll returns true if npa matches all node IPs (of npa's given family)
+func (npa *NodePortAddresses) MatchAll() bool {
+	return npa.matchAll
+}
+
+// GetNodeAddresses return all matched node IP addresses for npa's CIDRs. If no matching
+// IPs are found, it returns an empty list.
 // NetworkInterfacer is injected for test purpose.
 func (npa *NodePortAddresses) GetNodeAddresses(nw NetworkInterfacer) (sets.Set[string], error) {
-	uniqueAddressList := sets.New[string]()
-
-	// First round of iteration to pick out `0.0.0.0/0` or `::/0` for the sake of excluding non-zero IPs.
-	for _, cidr := range npa.cidrStrings {
-		if IsZeroCIDR(cidr) {
-			uniqueAddressList.Insert(cidr)
-			return uniqueAddressList, nil
```

Comment:
before with a ZeroCIDR it will just return that, now it will parse all the IPs on the node, shouldn't we shortcut as before to maintain the behavior?